src/test_application/activescenes/common_scene.cpp

```diff
@@ -57,16 +57,16 @@ void CommonTestScene::update_hierarchy_delete()
     SysHierarchy::update_delete_descendents(
             m_basic.m_hierarchy,
             std::cbegin(m_delete), std::cend(m_delete),
-            [this] (ActiveEnt ent)
+            [this] (ActiveEnt const ent)
     {
         m_deleteTotal.push_back(ent);
     });
 }
 
 void CommonTestScene::update_delete()
 {
-    auto first = std::cbegin(m_deleteTotal);
-    auto last = std::cend(m_deleteTotal);
+    auto const& first = std::cbegin(m_deleteTotal);
+    auto const& last = std::cend(m_deleteTotal);
```

Comment:
Is it intentional that these are references?

Reply:
Yes. Many data structures do not return by value from their .begin() or .end() functions. Taking the result by const& instead of "auto" potentially avoids a copy-constructor call.